mllib/src/main/scala/org/apache/spark/mllib/classification/LogisticRegression.scala

```diff
@@ -444,13 +444,13 @@ class LogisticRegressionWithLBFGS
         lr.setFitIntercept(addIntercept)
         lr.setMaxIter(optimizer.getNumIterations())
         lr.setTol(optimizer.getConvergenceTol())
+        // Determine if we should cache the DF
+        lr.setHandlePersistence(input.getStorageLevel == StorageLevel.NONE)
```

Comment:
`handlePersistence` should be specified by the user rather than inferred by the algorithm.

Reply:
`mlllib.LoR` do not expose `HandlePersistence` to users now, and I think it maybe better to keep it.